skimage/util/_backends.py

```diff
@@ -0,0 +1,127 @@
+import os
+import warnings
+
+
+def _entry_points(group):
+    # Support Python versions before 3.10, which do not let you
+    # filter groups directly.
+    all_entry_points = entry_points()
+    if hasattr(all_entry_points, "select"):
+        selected_entry_points = all_entry_points.select(group=group)
+    else:
+        selected_entry_points = all_entry_points.get(group, ())
+    return selected_entry_points
+
+
+def disable_dispatching():
+    """Determine if dispatching has been disabled by the user"""
+    no_dispatching = os.environ.get("SKIMAGE_NO_DISPATCHING", False)
+    if no_dispatching == "1":
+        return True
+    else:
+        return False
+
+
+@lru_cache
+def all_backends():
+    """List all installed backends and information about them"""
+    backends = {}
+    backends_ = _entry_points(group="skimage_backends")
+    backend_infos = _entry_points(group="skimage_backend_infos")
+
+    for backend in backends_:
+        backends[backend.name] = {"implementation": backend}
+        try:
+            info = backend_infos[backend.name]
+            # Double () to load and then call the backend information function
+            backends[backend.name]["info"] = info.load()()
+        except KeyError:
+            pass
+
+    return backends
+
+
+def dispatchable(func):
+    """Mark a function as dispatchable.
+
+    When a decorated function is called the installed backends are
+    searched for an implementation. If no backend implements the function
+    then the scikit-image implementation is used.
+    """
+    func_name = func.__name__
+    func_module = func.__module__
+
+    # If no backends are installed at all or dispatching is disabled,
+    # return the original function. This way people who don't care about it
+    # don't see anything related to dispatching
+    if disable_dispatching() or not all_backends():
+        return func
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # Backends are tried in alphabetical order, this makes things
+        # predictable and stable across runs. Might need a better solution
+        # when it becomes common that users have more than one backend
+        # that would accept a call.
+        for name in sorted(all_backends()):
+            backend = all_backends()[name]
+            # If we have a BackendInformation object we check if the
+            # function we are looking for is implemented in the backend
+            if "info" in backend:
+                if (
+                    f"{func_module}.{func_name}"
+                    not in backend["info"].supported_functions
+                ):
+                    continue
+
+            backend_impl = backend["implementation"].load()
+
+            # Allow the backend to accept/reject a call based on the function
+            # name and the values of the arguments
+            wants_it = backend_impl.can_has(
+                f"{func_module}.{func_name}", *args, **kwargs
+            )
```

Comment:
just a suggestion
```suggestion
            backend_allows = backend_impl.can_has(
                f"{func_module}.{func_name}", *args, **kwargs
            )
```

Reply:
One nitpick:  All of these should use `f"{func_module}:{func_name}"`, I think.  That is just a bit clearer, and allows for class methods to be specified cleanly (we may not need that now, but `:` is a typical spelling in Python for this already).
The `func_name` is the `__qualname__` which may contain `.`.

Reply:
Would align with the [Entry point spec](https://packaging.python.org/en/latest/specifications/entry-points/). :+1: on using `module:func`. 

I feel like there's another standard out there that uses `::` but not sure where I got that notion from. :thinking:

Reply:
Done. Now using `{module_name}:{func_name}` everywhere